aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,223 @@
+
+Forcing a Particular Format
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you would like to require the use of a particular response format, without resorting to using a return type like ``JsonResult``, you can apply the ``[Produces]`` filter. The ``[Produces]`` filter specifies a response format. Like most :doc:`/mvc/controllers/filters`, this can be applied at the action, controller, or global scope.
+
+.. code-block:: c#
+
+  [Produces("application/json")]
+  public class AuthorsController
+
+The ``[Produces]`` filter will force all actions within the ``AuthorsController`` to return JSON-formatted responses, even if other formatters were configured for the application and the client provided an ``Accept`` header requesting a different, available format. See :doc:`/mvc/controllers/filters` to learn more, including how to apply filters globally.
+
+Special Case Formatters
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Some special cases are implemented using built-in formatters. By default, ``string`` return types will be formatted as `text/plain` (`text/html` if requested via ``Accept`` header). This behavior can be removed by removing the ``TextOutputFormatter``. You remove formatters in the ``Configure`` method in *Startup.cs* (shown below). Actions that have a model object return type will return a 204 No Content response when returning ``null``. This behavior can be removed by removing the ``HttpNoContentOutputFormatter``. The following code removes the ``TextOutputFormatter`` and `HttpNoContentOutputFormatter``.
+
+.. code-block:: c#
+
+  services.AddMvc(options =>
+  {
+    options.OutputFormatters.RemoveType<TextOutputFormatter>();
+    options.OutputFormatters.RemoveType<HttpNoContentOutputFormatter>();
+  });
+  
+Without the ``TextOutputFormatter``, ``string`` return types return 406 Not Acceptable.
+
+Without the ``HttpNoContentOutputFormatter``, null objects are formatted using the configured formatter. For example, the JSON formatter will simply return a response with a body of ``null``, while the XML formatter will return an empty XML element with the attribute ``xsi:nil="true"`` set.
+
+Response Format URL Mappings
+----------------------------
+
+Clients can request a particular format as part of the URL, such as in the query string or part of the path, or by using a format-specific file extension such as .xml or .json. The mapping from request path should be specified in the route the API is using. For example:
+
+.. code-block:: c#
+
+  [FormatFilter]
+  public class ProductsController
+  {
+    [Route("[controller]/[action]/[id].{format?}")]
+    public Product GetById(int id)
+  
+This route would allow the requested format to be specified as an optional file extension. The ``[FormatFilter]`` attribute checks for the existence of the format value in the ``RouteData`` and will map the response format to the appropriate formatter when the response is created.
```

Comment:
I show how to specify the format in the table of examples. I'm honestly not sure where the set of allowable formats comes from. @javiercn can you comment? I found the implementation here https://github.com/aspnet/Mvc/tree/1.0.0-rc2/src/Microsoft.AspNetCore.Mvc.Core/Formatters but couldn't find where the set of FormatterMappings is populated.